File: sprint5/By lessons/lesson7code3/lesson7code3.kt
fun bigLeftRotation(a: Node): Node {
    // Задаём обозначения.
    val b = a.right
    val c = b!!.left
    val M = c!!.left
    val N = c.right

    // Переусыновляем вершины.
    a.right = M
    b.left = N
    c.left = a
    c.right = b

    // Корректируем высоты.
    a.height -= 2
    b.height -= 1
    c.height += 1

    // Возвращаем новый корень.
    return c
}

File: sprint5/By lessons/lesson7code3/lesson7code3.js
function bigLeftRotation(a) {
  const b = a.right;
  const c = b.left;
  const M = c.left;
  const N = c.right;

  a.right = M;
  b.left = N;
  c.left = a;
  c.right = b;

  a.height -= 2;
  b.height -= 1;
  c.height += 1;
  return c;
}
